Server/src/Server.Infrastructure/Persistence/Repositories/UserBookRepository.cs

```diff
@@ -0,0 +1,45 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Server.Domain.Entities;
+using Server.Domain.Entities.User;
+using Server.Domain.Repositories;
+
+namespace Server.Infrastructure.Persistence.Repositories;
+
+internal sealed class UserBookRepository : IUserBookRepository
+{
+    private readonly BookBookDbContext _dbContext;
+
+    public UserBookRepository(BookBookDbContext dbContext)
+    {
+        _dbContext = dbContext;
+    }
+
+    public void Add(UserBook userBook)
+    {
+        _dbContext.Add(userBook);
+    }
+
+    public async Task<UserBook?> FirstOrDefaultByIdsAsync(Guid bookId, Guid userId, CancellationToken cancellationToken = default)
+    {
+        return await _dbContext.UserBooks.FirstOrDefaultAsync(x => x.UserId == userId && x.BookId == bookId, cancellationToken);
+    }
+
+    public async Task<List<Book>> GetAllByUserId(Guid userId, CancellationToken cancellationToken = default)
+    {
+        return await _dbContext.UserBooks
+            .Include(x => x.Book)
+            .ThenInclude(x => x.Publisher)
+            .Include(x => x.Book)
```

Comment:
raczej nie trzeba za każdym razem tego `Include` robić

Reply:
właśnie niestety trzeba z tego co patrzyłem xd, ale fakt brzydko to wygląda
https://stackoverflow.com/questions/50897638/how-to-call-theninclude-twice-in-ef-core